amplify/backend/function/expressServer/src/controllers/portfolioController.js

```diff
@@ -293,10 +293,130 @@ catch (err) {
 }
 }
 
+const getLeaguePortfolio = async (req,res,next) => {
+  try{
+
+    // check that the userId and leagueIds can be cast to valid objectIds
+    if (mongoose.Types.ObjectId.isValid(req.params.userId) === false || 
+        mongoose.Types.ObjectId.isValid(req.params.leagueId) === false ){
+      // check that the stock ID is correct
+      res.status(404)
+      res.errormessage = 'No portfolio found'
+      return next(
+        new Error(
+          'No portfolio found'
+        )
+      )
+    }
+
+    //check if the token is their token and corresponds to the params
+    if(req.user.id !== req.params.userId){
```

Comment:
This check isn't needed if we get the userID from the auth middleware instead I think. The request will never get this far if unauthorised and middleware will catch this. If we remove userID then too from route like in suggestion wont have to check req.params.userId either.